feat(TItle Scen, Store Scen)

GaMaius opened the pull request:
# 변경점 👍
 Title과 Store 씬을 추가하고 서로 이동할 수 있도록 제작함.
또한 씬을 넘어가는 동안 없어지면 안되는 것들을 DontDestoryOnLoad에 포함시킴.

# 테스트 💻
 Title 씬에서 실행시키면 됩니다.

# 버그 🐞
- 게임 씬에서 나왔다가 다시 들어가면 참조 오류 등 버그가 일어남
- 애로우가 나가지 않아도 소리가 남
 
# 비고 ✏
### 게임 문제점
- ExpManage하는 코드가 따로 분리가 안되어있고 Player에 있음
- 보스 음악 소리가 너무 큼
- 음악이 안어울림
- 보스가 패턴이 없음
- 공격 애니메이션, 몬스터들의 걷기 모션이 없음
- 피격시 작동하는 타격감이 없음
- 코드에 중복되는 부분이 많음
- 쓸데없이 매번 실행(체크되는) 함수 혹은 조건이 있음
- 리팩토링이 필요함
- 이름 선언에 통일성이 없음(제각각인 대소문자 등)(스크립트, 태그 등등)

GaMaius committed:
feat(TItle Scen, Store Scen)

Title과 Store 씬을 추가하고 서로 이동할 수 있도록 제작함.
또한 씬을 넘어가는 동안 없어지면 안되는 것들을 DontDestoryOnLoad에 포함시킴

GaMaius committed:
feat(fonts)

한글 폰트 추가

GaMaius committed:
fix(Player)

씬이 변환될 때 Player가 중복 생성되는 것을 방지

hwanginseop commented:
![image](https://github.com/user-attachments/assets/0f0a70a8-42e6-4fd8-953a-49d955426f49)

인게임 사망 이후 상점 - 재시작 시 게임 실행이 안되네요

hwanginseop commented:
씬분리나 전역변수 관리는 전체적으로 깔끔하게 되어있어서 좋네요. 다만 EnemInfo가 저번 시간 이후로 scriptableObj를 활용한 방향으로 진행되었으면 좀더 좋았지 싶은 아쉬움이 있습니다. 과제 수고하였습니다. 상술한 이슈 해결하고 말씀해주세용

GaMaius committed:
feat(Soldier), fix(DontDestroyOnLoad)

원래 근접 적이었던 Cat을 지운 후 Soldier로 교체
게임에서 나갔을 때 객체들이 그대로 남아있는 버그를 수정

GaMaius committed:
fix(BossAI)

object pool이 아닌 Destroy가 적용되어 있는 코드 수정

<현재 문제점>
- 씬 전환시 DontDestroyOnLoad에 인스턴트 할당 방식으로 연결한 참조들이 전부 끊김
  -> 참조 방식을 변경할 예정
- 씬 전환과 게임 내 데이터를 각 객체에서 관리함
  -> GameManager과 SceneManger를 만들어 하나로 통합할 예정

GaMaius committed:
fix(PoolObj)
